fem/fespace.cpp

```diff
@@ -1632,434 +1862,772 @@ void FiniteElementSpace::Construct()
    // This method should be used only for non-NURBS spaces.
    MFEM_VERIFY(!NURBSext, "internal error");
 
+   // Variable order space needs a nontrivial P matrix + also ghost elements
+   // in parallel, we thus require the mesh to be NC.
+   MFEM_VERIFY(!IsVariableOrder() || Nonconforming(),
+               "Variable order space requires a nonconforming mesh.");
+
    elem_dof = NULL;
-   bdrElem_dof = NULL;
+   bdr_elem_dof = NULL;
    face_dof = NULL;
 
    ndofs = 0;
-   nedofs = nfdofs = nbdofs = 0;
+   nvdofs = nedofs = nfdofs = nbdofs = 0;
    bdofs = NULL;
-   fdofs = NULL;
+
    cP = NULL;
    cR = NULL;
+   cR_hp = NULL;
    cP_is_set = false;
    // 'Th' is initialized/destroyed before this method is called.
 
-   nvdofs = mesh->GetNV() * fec->DofForGeometry(Geometry::POINT);
+   int dim = mesh->Dimension();
+   int order = fec->GetOrder();
+
+   MFEM_VERIFY(mesh->GetNumGeometries(dim),
+               "Mesh was not correctly finalized.");
```

Comment:
This check caused autotest failures. It does not work with empty processors.

Reply:
Thank you, should hopefully be fixed now.

Reply:
By the way, this was triggered only when building with direct parallel solvers, like SuperLU_DIST -- from here: https://github.com/mfem/mfem/blob/33827fe5ccaccc5a6ae322c198ffd945ba0001dd/tests/unit/linalg/test_direct_solvers.cpp#L165
at least in 1D the mesh has 2 elements and we run the test on 4 ranks.